### ISSUE TITLE
Add installer for windows-arm64

### ISSUE THREAD

**Issue by pbo-linaro** [oneclick/rubyinstaller2#362]
Recently, this PR added Ruby support for windows-arm64: https://github.com/ruby/ruby/pull/8995
It fixes build and runtime issues.

Just after this, windows-arm64 MSYS2 ruby package was re-enabled thanks to this: https://github.com/msys2/MINGW-packages/pull/19179

Now, the missing piece for Ruby ecosystem is to create a windows-arm64 installer for RubyInstaller2.

What kind of work is expected to be done here?
Is it simply a packaging of various MSYS2 packages?

Thanks,
Pierrick

**Comment by pbo-linaro** [comment]
@mohits Sorry to ping you directly, but I noticed you were triaging some issues. Is there someone specifically we could add to this discussion?

**Comment by mohits** [comment]
Hi @pbo-linaro - I think most of the hard work is done by @larskanis and @MSP-Greg when it comes to handling the root causes of the problems.

I'm not fully plugged in to how the RubyInstaller2 is set up or built,  but help out with some triage, and follow-up questions and answers since I do want to help the #RubyOnWindows community if I can. That way, I can keep reducing the open issues or get some help across so that the others can focus on the hard work :)

**Comment by MSP-Greg** [comment]
Similar to the problem of how to provide non-Intel macOS builds for CI, windows-arm64 systems are not available on GitHub Actions.

I haven't looked around to see if any CI providers have them.

**Comment by pbo-linaro** [comment]
Thanks for your answers. Would you be open to use a self-hosted runner to fill this gap, or do you rely only on GitHub runners for now?

**Comment by johnnyshields** [comment]
@MSP-Greg @larskanis if the lack of a Github Actions runner is the limiting factor here, my team is happy to contribute a self-hosted Windows ARM runner until such time that Github officially supports it.

**Comment by MSP-Greg** [comment]
Just want to clarify one issue.  The `setup-ruby` action used with GitHub Actions has a system that provides build tools and/or packages when using Windows Rubies.  That system currently won't function on arm64.

This was done for two main reasons.

First of all, many repos run CI on Windows, but the maintainers are not familiar with Windows at the level needed to set up the required build system.   Currently, the code selects whether to install MINGW, UCRT, or mswin tools/packages, and also selects a proper OpenSSL package.  Note that the build systems are preinstalled on the Ubuntu & macOS runners.

Secondly, because `setup-ruby` handles `bundle install` and also caching of the bundled gems, the build tools need to be installed before `bundle install` is run.  Build tools are required for any extension gems that are not pre-compiled.

So, although adding arm64 builds to RubyInstaller2 is worthwhile, there will need to be changes to use those builds with GitHub Actions, regardless of whether the runners are self-hosted or not...

**Comment by johnnyshields** [comment]
@MSP-Greg in light of [this comment](https://github.com/ruby/setup-ruby/issues/600#issuecomment-2128482665), who should be adding support for Windows ARM first--Windows Installer (this repo) or setup-ruby?

**Comment by eregon** [comment]
RubyInstaller. setup-ruby nevwr built Ruby on Windows.
As @MSP-Greg mentioned indeed more work will be needed later in setup-ruby to get the toolchain working, but anyway RubyInstaller builds need to happen first.

**Comment by esimkowitz** [comment]
@MSP-Greg @larskanis now that Windows on ARM GitHub-hosted runners are available, is there anything else blocking setting this up?

**Comment by johnnyshields** [comment]
@esimkowitz Windows ARM64 runners are now GA, however they are still a paid offering:
https://github.blog/changelog/2024-09-03-github-actions-arm64-linux-and-windows-runners-are-now-generally-available/

**Comment by esimkowitz** [comment]
@johnnyshields correct. Looking at the Actions usage for this repo (<=20mins/runner/day), it would be <=$48/mo.

**Comment by pbo-linaro** [comment]
Expect them to be freely available for public repositories by the end of the year, as mentioned in this blog article: https://github.blog/news-insights/product-news/arm64-on-github-actions-powering-faster-more-efficient-build-systems/#get-started-using-arm-hosted-runners-today.

It will definitely help the windows-arm64 ecosystem to grow up!

**Comment by larskanis** [comment]
Thank you @pbo-linaro for great work in this area!

In the last year I started to port RubyInstaller to ARM and bough a Raspberry PI 4 to run Windows-11 on it. But it turned out to be way too slow and not stable enough to do proper development. I would like to continue this and future work, but I need a suitable hardware. That's why I started a fundraiser for a Windows ARM laptop. My promise is that I'll deliver a ARM64 native version in the next months.

If anyone is interested in supporting me then make a donation here: https://gofund.me/81b1f11f

**Comment by larskanis** [comment]
The [fundraiser](https://gofund.me/81b1f11f) above was successful, so that I've got a new Windows-11 device with ARM64 CPU. It's really fun to work with that notebook, so that I've got quite far with porting the RubyInstaller to ARM64: https://github.com/oneclick/rubyinstaller2/pull/399

**Comment by second-reality** [comment]
Thanks and congrats for your work on this @larskanis, it's a great achievement!

**Comment by pbo-linaro** [comment]
Thanks and congrats for your work on this @larskanis, it's a great achievement!

**Comment by larskanis** [comment]
Reopening this until RubyInstaller for ARM is actually released.

**Comment by marcpems** [comment]
@larskanis do you have a release date for the installer?

**Comment by larskanis** [comment]
@marcpems Release date is 2025-01-18. I'll add a release post on https://rubyinstaller.org with more details soon.

**Comment by larskanis** [comment]
The release post with some details is here: https://rubyinstaller.org/2025/01/19/rubyinstaller-3.4.1-2-released.html
